Documentation/platforms/container-linux-systemd.md

```diff
@@ -0,0 +1,138 @@
+
+## etcd v0, v2
+
+[Container Linux][container-linux-docs] filesystem includes `etcd.service` for v0.x and `etcd2.service` for v2.x, both of which will be removed in later versions. See [Container Linux FAQ][container-linux-faq] for more detail.
+
+To check if it exists:
+
+```
+systemctl cat etcd.service
+systemctl cat etcd2.service
+```
+
+To check if it is running:
+
+```
+systemctl status etcd.service
+systemctl status etcd2.service
+```
+
+To stop and disable:
+
+```
+systemctl stop etcd.service
+systemctl stop etcd2.service
+
+systemctl disable etcd.service
+systemctl disable etcd2.service
+```
+
+## etcd v3
+
+[Container Linux][container-linux-docs] filesystem does not include etcd v3 binaries, thus different versions of etcd v3 can be fetched via `etcd-member.service`.
+
+To check if it exists:
+
+```
+systemctl cat etcd-member.service
+```
+
+To check if it is running:
+
+```
+systemctl status etcd-member.service
+```
+
+And write a systemd drop-in unit to override the default Container Linux settings:
+
+```
+cat > /tmp/20-cl-etcd-member.conf <<EOF
+[Service]
+Environment="ETCD_IMAGE_TAG=v3.2.0"
+Environment="ETCD_DATA_DIR=/var/lib/etcd"
+Environment="ETCD_SSL_DIR=/etc/ssl/certs"
+Environment="ETCD_OPTS=--name s1 \
+    --listen-client-urls https://10.240.0.1:2379 \
+    --advertise-client-urls https://10.240.0.1:2379 \
+    --listen-peer-urls https://10.240.0.1:2380 \
+    --initial-advertise-peer-urls https://10.240.0.1:2380 \
+    --initial-cluster s1=https://10.240.0.1:2380,s2=https://10.240.0.2:2380,s3=https://10.240.0.3:2380 \
+    --initial-cluster-token mytoken \
+    --initial-cluster-state new \
+    --client-cert-auth \
+    --trusted-ca-file /etc/ssl/certs/etcd-root-ca.pem \
+    --cert-file /etc/ssl/certs/s1.pem \
+    --key-file /etc/ssl/certs/s1-key.pem \
+    --peer-client-cert-auth \
+    --peer-trusted-ca-file /etc/ssl/certs/etcd-root-ca.pem \
+    --peer-cert-file /etc/ssl/certs/s1.pem \
+    --peer-key-file /etc/ssl/certs/s1-key.pem \
+    --auto-compaction-retention 1"
+EOF
+mv /tmp/20-cl-etcd-member.conf /etc/systemd/system/etcd-member.service.d/20-cl-etcd-member.conf
+```
+
+To see all runtime drop-in changes for system units:
+
+```
+systemd-delta --type=extended
+```
+
+To enable and start:
+
+```
+systemctl daemon-reload
+systemctl enable etcd-member.service
+systemctl start etcd-member.service
+```
+
+To see the logs:
+
+```
+journalctl -u etcd-member.service
+```
+
+To stop and disable the service:
+
+```
+systemctl stop etcd-member.service
+systemctl disable etcd-member.service
+```
+
+## Provisioning etcd v3
+
+The cluster bootstrapping can be done easily with [Ignition][container-linux-ignition], where `coreos-metadata.service` dynamically fetches the machine's IP for discovery. And [Container Linux Config Transpiler][container-linux-ct] helps verify etcd configurations, while automatically generating the Ignition files:
```

Comment:
Cluster bootstrapping in Container Linux is simplest with [Ignition][container-linux-ignition];  `coreos-metadata.service` dynamically fetches the machine's IP for discovery. Note that etcd's discovery service protocol is only meant for bootstrapping, and cannot be used with runtime reconfiguration or cluster monitoring.

To avoid accidental misconfiguration, [Container Linux Config Transpiler][container-linux-ct] helpfully verifies etcd configurations when generating  Ignition files: